infrastructure/snyk_api/scan.go

```diff
@@ -0,0 +1,168 @@
+	} `json:"semver"`
+	PublicationTime string        `json:"publicationTime"`
+	DisclosureTime  string        `json:"disclosureTime"`
+	IsUpgradable    bool          `json:"isUpgradable"`
+	IsPatchable     bool          `json:"isPatchable"`
+	IsPinnable      bool          `json:"isPinnable"`
+	Identifiers     Identifiers   `json:"identifiers"`
+	Credit          []string      `json:"credit"`
+	CVSSv3          string        `json:"CVSSv3"`
+	CvssScore       float64       `json:"cvssScore"`
+	CvssDetails     []CvssDetails `json:"cvssDetails"`
+	UpgradePath     []string      `json:"upgradePath"`
+}
+
+type Org struct {
+	Name string `json:"name"`
+	Id   string `json:"id"`
+}
+
+type LicensesPolicy struct {
+	Severities struct {
+		AGPL10     string `json:"AGPL-1.0"`
+		AGPL30     string `json:"AGPL-3.0"`
+		Artistic10 string `json:"Artistic-1.0"`
+		Artistic20 string `json:"Artistic-2.0"`
+		CDDL10     string `json:"CDDL-1.0"`
+		CPOL102    string `json:"CPOL-1.02"`
+		EPL10      string `json:"EPL-1.0"`
+		GPL20      string `json:"GPL-2.0"`
+		GPL30      string `json:"GPL-3.0"`
+		LGPL20     string `json:"LGPL-2.0"`
+		LGPL21     string `json:"LGPL-2.1"`
+		LGPL30     string `json:"LGPL-3.0"`
+		MPL11      string `json:"MPL-1.1"`
+		MPL20      string `json:"MPL-2.0"`
+		MSRL       string `json:"MS-RL"`
+		SimPL20    string `json:"SimPL-2.0"`
+	} `json:"severities"`
+}
+
+type ScanResult struct {
+	Ok     bool `json:"ok"`
+	Issues struct {
+		Vulnerabilities []Vulnerability `json:"vulnerabilities"`
+	} `json:"issues"`
+	DependencyCount int            `json:"dependencyCount"`
+	Org             Org            `json:"org"`
+	LicensesPolicy  LicensesPolicy `json:"licensesPolicy"`
+	PackageManager  string         `json:"packageManager"`
+}
+
+func (s *SnykApiClientImpl) ScanPackage(packageName string, packageVersion string, packageManager string) (*ScanResult, error) {
+	method := "ScanPackage"
+
+	log.Debug().Str("method", method).
+		Str("packageManager", packageManager).
+		Str("packageName", packageName).
+		Str("packageVersion", packageVersion).
+		Msg("API: Scanning package")
+	defer log.Debug().Str("method", method).
+		Str("packageManager", packageManager).
+		Str("packageName", packageName).
+		Str("packageVersion", packageVersion).
+		Msg("API: Scanning package - DONE")
+	c := config.CurrentConfig()
+	utm := fmt.Sprintf("utm_source=snyk-ls-%s-(%s:%s)&utm_medium=IDE&type=json", config.Version, c.IntegrationName(), c.IntegrationVersion())
+
+	// default url
+
+	var scanPackageURL string
+	switch packageManager {
+	case "sbt":
+		fallthrough
+	case "gradle":
+		fallthrough
+	case "maven":
+		if !strings.Contains(packageName, ":") {
+			return nil, fmt.Errorf("API: invalid package name %s", packageName)
+		}
+		group := strings.Split(packageName, ":")[0]
+		artifactId := strings.Split(packageName, ":")[1]
+		scanPackageURL = fmt.Sprintf("/test/%s/%s/%s/%s?%s", packageManager, group, artifactId, packageVersion, utm)
```

Comment:
Question: Shouldn't there be a 'v1' prefix? As to be able to use 'rest' calls as well.